### PR TITLE
Completed info dialog

### PR DESCRIPTION
**Pull request by YangQi007**

### CODE REVIEW

**Review by anderspitman — CHANGES_REQUESTED** [review]
One small change

File: histogram.js [hunk 1]
```diff
@@ -118,14 +120,30 @@ class HistogramElement extends HTMLElement {
...
     }
-    else {
-      (async () => {
-        const data = await getDataFromAttr(this);
-        if (data) {
-          this._histo.update(data);
+
+    const renderHistogramWhenVisible = () => {
+        if (!useBroker) {
+          (async () => {
```
Comment (anderspitman) [inline histogram.js:126]: You can simplify this a bit by making `renderHistogramWhenVisible` async and removed the inline function

**Review by anderspitman — CHANGES_REQUESTED** [review]
This is working beautifully! Just one small detail.

File: histogram.js [hunk 2]
```diff
@@ -118,14 +120,32 @@ class HistogramElement extends HTMLElement {
...
-    else {
-      (async () => {
-        const data = await getDataFromAttr(this);
-        if (data) {
-          this._histo.update(data);
+
+    const renderHistogramWhenVisible = async () => {
+      if (!useBroker) {
+          try {
+              const data = await getDataFromAttr(this);
```
Comment (anderspitman) [inline histogram.js:127]: Can you fix the indentation here? It should be 2 spaces. We should use 2 space in general for web apps..

**Review by anderspitman — CHANGES_REQUESTED** [review]

File: histogram.js (hunk 2 above)
Comment (YangQi007) [reply histogram.js]: I fixed it! Please recheck it.

**Comment by anderspitman** [comment]
Great work!

**Pull request merged**